packages/legend-studio/src/components/editor/edit-panel/mapping-editor/MappingTestEditor.tsx

```diff
@@ -333,6 +356,57 @@ export const MappingTestRelationalInputDataBuilder = observer(
   },
 );
 
+export const MappingTestInputDataTypeBuilder = observer(
+  (props: {
+    inputDataState:
+      | MappingTestObjectInputDataState
+      | MappingTestRelationalInputDataState;
+    isReadOnly: boolean;
+  }) => {
+    const { inputDataState, isReadOnly } = props;
+    const inputTypeList: string[] = [];
+    const chosenInputType = inputDataState.inputData.inputType;
+
+    if (inputDataState instanceof MappingTestObjectInputDataState) {
+      inputTypeList.push(ObjectInputType.JSON);
+    } else {
```

Comment:
instead of `else`, let's do another case for relational here, this would be bad for things like CSV

Reply:
We already check that before, here we only allow processing object and relational input data state